### PR TITLE
[search] Add IsStreetSynonymWithMisprints method.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

File: indexer/search_string_utils.cpp [hunk 1]
```diff
@@ -283,8 +320,42 @@ class StreetsSynonymsHolder
...
+      auto const p = q.front();
+      q.pop();
+
+      auto const & trieIt = p.first;
+      auto const & dfaIt = p.second;
+
+      if (dfaIt.Accepts())
+        return true;
+
+      trieIt.ForEachMove([&q, &dfaIt](UniChar const & c, TrieIt const & curDfaIt) {
```
Comment (mpimenov) [inline indexer/search_string_utils.cpp]: s/trieIt/curTrieIt/
s/curDfaIt/nextTrieIt/

Comment (mpimenov) [reply indexer/search_string_utils.cpp]: and rename other iterator-related variable too, please

Comment (tatiana-yan) [reply indexer/search_string_utils.cpp]: done

File: indexer/search_string_utils.cpp [hunk 2]
```diff
@@ -283,8 +320,42 @@ class StreetsSynonymsHolder
...
+    queue<State> q;
+    q.emplace(trieRoot, dfa.Begin());
+
+    while (!q.empty())
+    {
+      auto const p = q.front();
+      q.pop();
+
+      auto const & currTrieIt = p.first;
+      auto const & curDfaIt = p.second;
```
Comment (mpimenov) [inline indexer/search_string_utils.cpp]: s/cur/curr/

Comment (tatiana-yan) [reply indexer/search_string_utils.cpp]: done

**Pull request merged**